### ISSUE TITLE
shell escape problems

### ISSUE THREAD

**Issue by jungle-boogie** [numcommand/num#7]
Hello,

It appears there's some kind of shell escape problem occurring with these versions of num:
a5f396229ec2cabec2d149db91f7f77e31071bd5
1511f9804188f18c80ac2b03366d6f6671867fe9
b5faf1e0ff3aa891068778352c3a4d8899c87d70
959ac74c43fad4bea9b2cb3ee9254f986e20ad58
f8068fda21f5c26b41fc1dd5c85eb32761586380
a86fd69844324abf97703e17d3ad860629fb50c7
fa7da18555f6ce8ef09fc36a91652a52f3e1f51f

Problem does not exist with this commit:
42d55d98c92170d7da939b0ca9d602a9495a9018

Data:
https://clbin.com/O7b1o

broken:

```
% cat data | num sum
1.79559e+06% 
```

Working:

```
% cat data | ./42d55 sum 
1.79559e+06
```

Notice the broken example contains a `%` at the end. If I were using the `/bin/sh` shell, there would be a `$` at the end. Both `%` and `$` are not expected in the results.

**Comment by joelparkerhenderson** [comment]
Thanks for the report and details. Fixed in version 1.2.1 which is ready now.

(The error was due to the output formatting features, when Num changed from using `print` to `printf` for output, so the final newline wasn't printing. You saw `%` and `$` because your shell was telling you the Num output was complete, but was missing a final newline.)

**Comment by jungle-boogie** [comment]
Hi @joelparkerhenderson,

Thanks for correcting the problem and the explanation. I should have taken the time to review the changes but I'm not sure what I would have know to look for.

Thanks a million for `num`!
